projects/plugins/boost/app/modules/cache/pre-wordpress/Boost_Cache_Utils.php

```diff
@@ -182,4 +208,16 @@ public static function is_visible_post_type( $post ) {
 		}
 		return true;
 	}
+
+	/**
+	 * Given a request_uri and its parameters, return the filename to use for this cached data. Does not include the file path.
+	 *
+	 * @param array  $parameters  - An associative array of all the things that make this request special/different. Includes GET parameters and COOKIEs normally.
+	 */
+	public static function get_request_filename( $parameters ) {
+
+		$key_components = apply_filters( 'boost_cache_key_components', $parameters );
+
+		return md5( json_encode( $key_components ) ) . '.html'; // phpcs:ignore WordPress.WP.AlternativeFunctions.json_encode_json_encode
```

Comment:
(judging by the path) this is too early to use `wp_json_encode`? I wonder if we should use JSON Encode flags here to ensure that the JSON that's generated is consistent.

Reply:
Yeah. It is too early. Should be ok though, as it's passed to md5 immediately, and the cookie/get arrays are cached and don't change. When saving a new cache file the md5 hash returned will be the same, as at the start of the process.